app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,71 @@
+class TasksController < ApplicationController
+  #write a list of tasks in a hash
+  # TASKS =[{id: 1, thing:"Go for a walk", status: "incomplete"}, {id: 2, thing:"groceries", status: "incomplete"}, {id: 3, thing:"Do this task builder project", status: "incomplete"}, {id: 4, thing:"lightening talk", status: "incomplete"}]
+
+  #write an index function that returns
+
+  def index
+    @tasks = Task.all.order(:id)
+  end
+
+  def show
+    @task = Task.find(params[:id].to_i)
```

Comment:
It's is best-practice to have an action ready if you can't find the task represented by the id.  This applies to `edit`, `create`, `update`, `delete`, and `mark_unmark` actions.